### PR TITLE
Expand init prompts in template

### LINKED ISSUE #728: [ADAP-786] Hints for `dbt init` auto-profile completion don't display in as helpful a way as they could
The hints referenced here:

https://github.com/dbt-labs/dbt-snowflake/blob/fd28b35263b3e3ee1fac43e2351ed02c8f3b02a5/dbt/include/snowflake/profile_template.yml#L4-L5

Look a little confusing in practice.

Screenshot 2023-08-08 at 3 32 46 PM

I think making them more descriptive like 'What is your account? [The value here <account>.snowflake-computing.com]' etc will help the init flow.

### PR DESCRIPTION
**Pull request by gwenwindflower**
resolves #728 

### Problem

Prompts were very terse and didn't display great in the CLI.

### Solution

Made 'em more verbose.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: dbt/include/snowflake/profile_template.yml [hunk 1]
```diff
@@ -2,33 +2,33 @@ fixed:
   type: snowflake
 prompts:
   account:
-    hint: 'https://<this_value>.snowflakecomputing.com'
+    hint: 'What is your account id for Snowflake? [https://<this_value>.snowflakecomputing.com]'
   user:
-    hint: 'dev username'
+    hint: 'What is your personal user login for Snowflake?'
```
Comment (mikealfare) [inline dbt/include/snowflake/profile_template.yml:7]: I feel like we should either go with `personal user` here and `personal password` below, or just `user` here and `password` below.

File: dbt/include/snowflake/profile_template.yml [hunk 2]
```diff
@@ -2,33 +2,33 @@ fixed:
...
   _choose_authentication_type:
     password:
       password:
-        hint: 'dev password'
+        hint: 'What is your password for Snowflake?'
         hide_input: true
     keypair:
       private_key_path:
-        hint: 'path/to/private.key'
+        hint: 'What is the path/to/private.key for Snowflake authentication?'
```
Comment (mikealfare) [inline dbt/include/snowflake/profile_template.yml:15]: I like that we're making these more "human" hints, but then the path example should probably be updated to reflect that grammatically; e.g. `path/to/the/private.key`.

**Review by mikealfare — COMMENTED** [review]
I left a few comments, and then we would need a changie entry as well. I think I would call this Under the Hood?

**Comment by dbeatty10** [comment]
> I left a few comments, and then we would need a changie entry as well. I think I would call this Under the Hood?

@gwenwindflower [here's](url) the instructions to create an changelog entry:
1. Follow the steps to [install changie](https://changie.dev/guide/installation/) for your system.
1. `changie new`
    -  Fill out the answers
    -  It will spit out a `.yml` file at the end
1. Commit and push that `.yml` file. I typically use a simple commit message like "Changelog entry".

File: dbt/include/snowflake/profile_template.yml (hunk 1 above)
Comment (dbeatty10) [inline dbt/include/snowflake/profile_template.yml:7]:
```suggestion
    hint: 'What is your user login for Snowflake?'
```

File: dbt/include/snowflake/profile_template.yml (hunk 2 above)
Comment (dbeatty10) [inline dbt/include/snowflake/profile_template.yml:15]:
```suggestion
        hint: 'What is the path/to/the/private.key for Snowflake authentication?'
```

File: dbt/include/snowflake/profile_template.yml (hunk 1 above)
Comment (dbeatty10) [reply dbt/include/snowflake/profile_template.yml]: Resolved via https://github.com/dbt-labs/dbt-snowflake/pull/729/commits/58ce8f2838a9f9b91c11c2cb3d92aa568c09d25a

File: dbt/include/snowflake/profile_template.yml (hunk 2 above)
Comment (dbeatty10) [reply dbt/include/snowflake/profile_template.yml]: Resolved via https://github.com/dbt-labs/dbt-snowflake/pull/729/commits/58ce8f2838a9f9b91c11c2cb3d92aa568c09d25a

**Comment by mikealfare** [comment]
In order to avoid some extra work, we're going to wait until the adapter/core decoupling work is done, and then update this PR with those changes (should be virtually nothing) before merging in.

**Pull request closed without merging**